src/EntityFramework6.Npgsql/NpgsqlTextFunctions.cs

```diff
@@ -368,5 +369,38 @@ public static string TsRewrite(string query, string target, string substitute)
         {
             throw new NotSupportedException();
         }
-    }   
+
+        /// <summary>
+        /// Matches regular expression, case insensitive. Generates the "~*" operator.
+        /// http://www.postgresql.org/docs/current/static/functions-matching.html#FUNCTIONS-POSIX-REGEXP
+        /// </summary>
+        /// <param name="input">A string to search for a match.</param>
+        /// <param name="pattern">A regular expression pattern to match.</param>
+        /// <returns>
+        /// <see langword="true"/> if <paramref name="input"/> matches <paramref name="pattern"/>,
+        /// otherwise, <see langword="false"/>.
+        /// </returns>
+        [DbFunction("Npgsql", "regex_is_match")]
+        public static bool RegexIsMatch(string input, string pattern)
```

Comment:
I think that if this (and its overload) were named MatchRegex it would be consistent the "Match" method which generates the @@ operator.